dynatrace_topology/stackstate_checks/dynatrace_topology/dynatrace_topology.py

```diff
@@ -0,0 +1,276 @@
+
+    def collect_relations(self, component, externalId):
+        """
+        Collects relationships from different component-types
+        :param component: the component for which relationships need to be extracted and processed
+        :param externalId: the component externalId for and from relationship will be created
+        :return: None
+        """
+        outgoing_relations = component.get("fromRelationships", {})
+        incoming_relations = component.get("toRelationships", {})
+        for relation_type in outgoing_relations.keys():
+            # Ignore `isSiteOf` relation since location components are not processed right now
+            if relation_type != "isSiteOf":
+                for target_id in outgoing_relations[relation_type]:
+                    self.relation(externalId, target_id, relation_type, {})
+        for relation_type in incoming_relations.keys():
+            # Ignore `isSiteOf` relation since location components are not processed right now
+            if relation_type != "isSiteOf":
+                for source_id in incoming_relations[relation_type]:
+                    self.relation(source_id, externalId, relation_type, {})
+
+    def filter_data(self, data):
+        """
+        Delete the un-necessary relationships from the data
+        """
+        if "fromRelationships" in data:
+            del data["fromRelationships"]
+        if "toRelationships" in data:
+            del data["toRelationships"]
+        return data
+
+    def collect_processes(self):
+        """
+        Collects the response from the Dynatrace Process API endpoint
+        """
+        endpoint = self.url + "/api/v1/entity/infrastructure/processes"
+        processes = self.get_json_response(endpoint)
+        return processes
+
+    def collect_hosts(self):
+        """
+        Collects the response from the Dynatrace Host API endpoint
+        """
+        endpoint = self.url + "/api/v1/entity/infrastructure/hosts"
+        hosts = self.get_json_response(endpoint)
+        return hosts
+
+    def collect_applications(self):
+        """
+        Collects the response from the Dynatrace Application API endpoint
+        """
+        endpoint = self.url + "/api/v1/entity/applications"
+        applications = self.get_json_response(endpoint)
+        return applications
+
+    def collect_process_groups(self):
+        """
+        Collects the response from the Dynatrace Process-Group API endpoint
+        """
+        endpoint = self.url + "/api/v1/entity/infrastructure/process-groups"
+        process_groups = self.get_json_response(endpoint)
+        return process_groups
+
+    def collect_services(self):
+        """
+        Collects the response from the Dynatrace Service API endpoint
+        """
+        endpoint = self.url + "/api/v1/entity/services"
+        services = self.get_json_response(endpoint)
+        return services
+
+    def process_component(self, response, component_type):
```

Comment:
You are actually processing both components and relations, hence topology. `process_topology`?